lib/nform/builder.rb

```diff
@@ -102,6 +102,10 @@ def number_field(k, label: nil, default: nil)
       zjoin label_for(k,text:label), input_for(k,type:'number',default:default,pattern:'\d*'), error_for(k)
     end
 
+    def natural_number_field(k, label: nil, default: nil)
+      zjoin label_for(k,text:label), input_for(k,type:'number',default:default,pattern:'\d*',min:0), error_for(k)
```

Comment:
See the the `input_for` method called here, in `input_for` above it calls `tag` which can take any old hash of attributes for the tag. I agree with your input in the comments, it's probably cleaner if the method call was just `input_field key, type: "text", **args` and then this can be called with whatever options are situationally appropriate